enterprise/coderd/workspaceproxy.go

```diff
@@ -122,9 +137,16 @@ func (api *API) postWorkspaceProxy(rw http.ResponseWriter, r *http.Request) {
 
 	aReq.New = proxy
 	httpapi.Write(ctx, rw, http.StatusCreated, codersdk.CreateWorkspaceProxyResponse{
-		Proxy:      convertProxy(proxy),
+		Proxy: convertProxy(proxy, proxyhealth.ProxyStatus{
+			Proxy:     proxy,
+			Status:    proxyhealth.Unregistered,
+			CheckedAt: time.Now(),
+		}),
 		ProxyToken: fullToken,
 	})
+
+	// Update the proxy health cache to include this new proxy.
+	go api.forceWorkspaceProxyHealthUpdate(api.ctx)
```

Comment:
This won't do anything since it's not registered yet.

Reply:
It adds the "Unregistered" to the cache. Before you call this, the proxy is not in the cache since the cache pulls it's list from the database.

So subsequent calls to proxy will now return this unregistered.